pkg/scheduler/framework/plugins/registry.go

```diff
@@ -39,21 +42,54 @@ import (
 	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
 )
 
+type registryOptions struct {
+	ignoredResources []string
+}
+
+// Option configures a registry
+type Option func(*registryOptions)
+
+func WithIgnoredResources(r ...string) Option {
+	return func(o *registryOptions) {
+		o.ignoredResources = r
+	}
+}
+
 // NewInTreeRegistry builds the registry with all the in-tree plugins.
 // A scheduler that runs out of tree plugins can register additional plugins
 // through the WithFrameworkOutOfTreeRegistry option.
-func NewInTreeRegistry() framework.Registry {
+func NewInTreeRegistry(opts ...Option) framework.Registry {
+	var options registryOptions
+	for _, opt := range opts {
+		opt(&options)
+	}
 	return framework.Registry{
-		defaultpodtopologyspread.Name:              defaultpodtopologyspread.New,
-		imagelocality.Name:                         imagelocality.New,
-		tainttoleration.Name:                       tainttoleration.New,
-		nodename.Name:                              nodename.New,
-		nodeports.Name:                             nodeports.New,
-		nodepreferavoidpods.Name:                   nodepreferavoidpods.New,
-		nodeaffinity.Name:                          nodeaffinity.New,
-		podtopologyspread.Name:                     podtopologyspread.New,
-		nodeunschedulable.Name:                     nodeunschedulable.New,
-		noderesources.FitName:                      noderesources.NewFit,
+		defaultpodtopologyspread.Name: defaultpodtopologyspread.New,
+		imagelocality.Name:            imagelocality.New,
+		tainttoleration.Name:          tainttoleration.New,
+		nodename.Name:                 nodename.New,
+		nodeports.Name:                nodeports.New,
+		nodepreferavoidpods.Name:      nodepreferavoidpods.New,
+		nodeaffinity.Name:             nodeaffinity.New,
+		podtopologyspread.Name:        podtopologyspread.New,
+		nodeunschedulable.Name:        nodeunschedulable.New,
+		noderesources.FitName: func(plArgs *runtime.Unknown, handle framework.FrameworkHandle) (framework.Plugin, error) {
```

Comment:
the alternative (which I prefer) is to get rid of `FitArgs` altogether and only ignore what is deduced from the extenders.

Reply:
Let's remove `FitArgs`

Reply:
done